app/models/friendship.rb

```diff
@@ -0,0 +1,15 @@
+class Friendship < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :friend, class_name: "User"
+
+  validates :user, :friend, presence: true
+
+  def self.current_friend_ids(current_user)
```

Comment:
This should be on user and not take a user as a parameter.

On user this could be something like
```
def outstanding_requests
  friendships.where(approved: false)
end
```


Ultimately if you want users who user 1 hasn't sent a request to you could do something like 
```
select * from users u 
  where not exists (select 1 from friendships f
   where f.friend_id = u.id and f.user_id = 1);
```